src/index.js

```diff
@@ -107,9 +111,15 @@ export default class Timeline extends Component {
       }
 
       $el.setOptions(updatedOptions)
+      $el.setSelection(selection)
 
     } else {
-      $el = this.TimelineElement = new vis.Timeline(container, timelineItems, timelineGroups, options)
+      if (timelineGroups.length > 0) {
```

Comment:
as above, although we could probably contain this check in a variable and then just use that:

```js
const hasGroups = timelineGroups.length

if (hasGroups) { ... }
```